### PR TITLE
Add init(light:dark:) to NSColor and UIColor

### PR DESCRIPTION
**Pull request by MaxHaertwig**
This extension allows to create an NSColor or UIColor with different colors for light mode and dark mode. I created separate `NSColorExtensions.swift` and `UIColorExtensions.swift` files, because NSColor and UIColor use different APIs for this problem and can't be unified.

I couldn't figure out how to create a test for macOS, because it uses NSAppearance instead of UITraitCollection. Any suggestions?

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/SwifterSwift/UIKit/UIColorExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+//
+//  UIColorExtensions.swift
+//  SwifterSwift
+//
+//  Created by Max Haertwig on 10/06/19.
+//  Copyright © 2019 SwifterSwift
+//
+
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/UIKit/UIColorExtensions.swift]: Missing #if canImport(UIKit)

Comment (MaxHaertwig) [reply Sources/SwifterSwift/UIKit/UIColorExtensions.swift]: Thanks

**Comment by MaxHaertwig** [comment]
I'd like to add a test for `NSColor`, but I don't know how to set an NSAppearance without having a view hierarchy. Any suggestions?

**Comment by guykogus** [comment]
@MaxHaertwig I don't think there's any way around the view hierarchy. I ran this in Playgrounds and it works:
```Swift
let color = NSColor(light: .red, dark: .blue)
let view = NSView()
view.wantsLayer = true

NSAppearance.current = NSAppearance(named: .aqua)
view.layer?.backgroundColor = color.cgColor
view.layer?.backgroundColor == NSColor.red.cgColor

NSAppearance.current = NSAppearance(named: .darkAqua)
view.layer?.backgroundColor = color.cgColor
view.layer?.backgroundColor == NSColor.blue.cgColor
```

**Comment by MaxHaertwig** [comment]
Thanks @guykogus, I added a test.

File: Tests/AppKitTests/NSColorExtensionsTests.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
+//
+//  UIColorExtensionsTests.swift
+//  SwifterSwift
+//
+//  Created by Max Haertwig on 10/08/19.
+//  Copyright © 2019 SwifterSwift
+//
+
+import Foundation
```
Comment (guykogus) [inline Tests/AppKitTests/NSColorExtensionsTests.swift]: Shouldn't it be the same as the source file?
```Swift
#if canImport(Cocoa)
import Cocoa
```

Comment (MaxHaertwig) [reply Tests/AppKitTests/NSColorExtensionsTests.swift]: Of course

**Comment by MaxHaertwig** [comment]
@guykogus, @LucianoPAlmeida do you know why the travis build failes?

**Comment by LucianoPAlmeida** [comment]
Seems something on Danger, I'll take a look  as soon as possible :))

**Comment by guykogus** [comment]
@MaxHaertwig check that `NSColorExtensions.swift` was added to the `SwifterSwift-macOS` target?

**Comment by MaxHaertwig** [comment]
It belongs to `SwifterSwift-macOS `.

**Comment by MaxHaertwig** [comment]
I fixed the error. However, I don't know why codecov is complaining. Any ideas?

**Comment by guykogus** [comment]
@MaxHaertwig codecov can be a bit strange sometimes. I wouldn't worry about it. It's complaining about a decrease in code coverage from a recent `UIImage` extension.

**Comment by MaxHaertwig** [comment]
Is it possible to run the Travis build on a macOS 10.15 machine?

**Comment by guykogus** [comment]
@MaxHaertwig not yet https://docs.travis-ci.com/user/reference/osx/

**Comment by VincentSit** [comment]
@LucianoPAlmeida @guykogus 

I think this method should be changed to the following to be more compatible.
```swift
convenience init?(light: UIColor?, dark: UIColor?) {
    guard let light = light, let dark = dark else {
      return nil
    }
    
    if #available(iOS 13.0, *) {
      self.init(dynamicProvider: { $0.userInterfaceStyle == .dark ? dark : light })
    } else {
      self.init(cgColor: light.cgColor)
    }
  }
```

Otherwise you will encounter such errors in projects with a minimum deployment version lower than iOS 13.

image

The reason for using optional variables is to be compatible with the colors loaded from the bundle.

**Comment by guykogus** [comment]
What does your code look like that the colours would be optional?
The main issue with changing this to be a nullable initialiser is that it's a breaking change. You could just make `dark` optional, which may help and still keeps things simple.
```Swift
convenience init(light: UIColor, dark: UIColor?) {
    if #available(iOS 13.0, tvOS 13.0, *),
        let dark = dark {
        self.init(dynamicProvider: { $0.userInterfaceStyle == .dark ? dark : light })
    } else {
        self.init(cgColor: light.cgColor)
    }
}
```

**Comment by VincentSit** [comment]
@guykogus 

Because I use R.swift, loading the color from the bundle (via `init(named:)`) will return an optional value. Your suggestion is great, I did not expect it to cause a breaking change.

**Comment by guykogus** [comment]
My suggestions are usually great 😏 (that's a lie, never listen to me)
Feel free to create a PR 🙂

**Comment by VincentSit** [comment]
😆 Thank you, will do it shortly.

**Pull request merged**